### PR TITLE
feat: add support for HTML semantics

### PR DESCRIPTION
**Pull request by manekinekko**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes/features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature (please, look at the "Scope of the project" section in the README.md file)
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?

- [ ] Yes
- [ ] No

### CODE REVIEW

**Comment by manekinekko** [comment]
- [x] attach an HTML element to a selected layer
- [ ] add Codegen.

File: src/app/app.module.ts [hunk 1]
```diff
@@ -62,10 +62,14 @@ const StoreDebugModule = [
      * ENABLING THIS, WILL THROW: TypeError: Cannot assign to read only property 'microTask' of object '[object Object]'
      * See similar issue in NgRx: https://github.com/brandonroberts/ngrx-store-freeze/issues/17
      */
-    developmentMode: !environment.production
+    // developmentMode: !environment.production
```
Comment (Jefiozie) [inline src/app/app.module.ts]: Can this be removed?

File: src/app/app.module.ts [hunk 2]
```diff
@@ -91,7 +95,7 @@ export function HttpLoaderFactory(http: HttpClient) {
     }),
     RouterModule.forRoot(routes, {
       useHash: true,
-      enableTracing: !environment.production,
+      // enableTracing: !environment.production,
```
Comment (Jefiozie) [inline src/app/app.module.ts]: Can this be removed?

File: src/app/core/state/ui.state.ts [hunk 3]
```diff
@@ -107,6 +107,11 @@ export class InformUser {
   ) {}
 }
 
+export class LayerHtmlCustomTage {
```
Comment (Jefiozie) [inline src/app/core/state/ui.state.ts]:
```suggestion
export class LayerHtmlCustomTags {
```

File: src/app/core/state/ui.state.ts [hunk 4]
```diff
@@ -259,6 +264,18 @@ export class UiState {
     );
   }
 
+  @Action(LayerHtmlCustomTage)
```
Comment (Jefiozie) [inline src/app/core/state/ui.state.ts]:
```suggestion
  @Action(LayerHtmlCustomTags)
```

File: src/app/editor/preview/layer-settings/settings-html-mapper/settings-html-mapper.component.ts [hunk 5]
```diff
@@ -0,0 +1,176 @@
...
+        }
+      }
+    } else {
+      // reset state
+      this.htmlMapperGroupAutocomplete = this.stateGroups;
+      this.selectedHtmlElementDescription = '';
+      this.autocompletePanel.closePanel();
+      this.store.dispatch(new LayerHtmlCustomTag(null));
+    }
+  }
```
Comment (Jefiozie) [inline src/app/editor/preview/layer-settings/settings-html-mapper/settings-html-mapper.component.ts]: Do you think we should refactor this to separate functions with correct names? For example, `resetstate` can be a function for line 129-133. It think, by doing this we have a better understanding of the goal of this code.

File: src/app/editor/preview/layer-settings/settings-html-mapper/settings-html-mapper.component.ts [hunk 6]
```diff
@@ -0,0 +1,176 @@
...
+        .filter(group => group.elements.length > 0);
+    }
+
+    return this.stateGroups;
+  }
+
+  private find(value: string): XLayersWebCodeGenTag {
+    if (value) {
+      const vals = this.stateGroups.flatMap(group => [...group.elements]);
+      return vals.find(item => item.name === value);
```
Comment (Jefiozie) [inline src/app/editor/preview/layer-settings/settings-html-mapper/settings-html-mapper.component.ts]:
```suggestion
      const elements = this.stateGroups.flatMap(group => [...group.elements]);
      return elements.find(element=> element.name === value);
```

File: src/tsconfig.spec.json [hunk 7]
```diff
@@ -3,8 +3,8 @@
   "compilerOptions": {
     "outDir": "../out-tsc/spec",
     "module": "commonjs",
-    "types": ["jest", "node", "sketchapp"]
+    "types": ["jest", "node"]
   },
   "files": ["polyfills.ts"],
-  "include": ["**/*.spec.ts", "**/*.d.ts"]
+  "include": ["**/*.spec.ts", "**/*.d.ts", "src/@types/sketchapp/index.d.ts"]
```
Comment (Jefiozie) [inline src/tsconfig.spec.json]: Is this temporarily?

**Pull request closed without merging**